src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -431,12 +431,8 @@ private static JsonTypeInfo<Queue<TValue>> CreateQueueTypeInfo<TValue>(JsonTypeI
             Debug.Assert(queueTypeInfo.IsConfigured);
             JsonSerializerOptions options = queueTypeInfo.Options;
             var bufferState = new ReadBufferState(options.DefaultBufferSize);
-            ReadStack readStack = new ReadStack
-            {
-                SupportContinuation = true
-            };
-
-            readStack.Initialize(queueTypeInfo);
```

Comment:
Alternative workaround:
```csharp
ReadStack readStack = default;
readStack.SupportContinuation = true
```

```

Reply:
I'm just reverting to what we had previously here.

I'm still curious though, would have expected that `ReadStack readStack = new ReadStack();` is equivalent to `ReadStack readStack = default;` when it comes to structs.

Reply:
it is. The problem is in the object initialization syntax - Roslyn creates a struct copy, see 
![image](https://user-images.githubusercontent.com/523221/182668518-b83ff4ab-b6b5-4639-ae33-cf84b68f44d7.png)

and since the struct is quite huge such a copy is expensive.

Reply:
It's tracked by https://github.com/dotnet/runtime/issues/8887

Reply:
![image](https://user-images.githubusercontent.com/523221/182669120-53d4beb9-d454-449c-8a59-039af623cb2f.png)

Reply:
Interesting, thanks. Would have assumed that it simply decompiles to what `Good` is doing.

Reply:
> It's tracked by https://github.com/dotnet/runtime/issues/8887

Maybe change Roslyn's behaviour instead?

Reply:
> > It's tracked by #8887
> 
> Maybe change Roslyn's behaviour instead?

no, it's a correct IL and such patterns might appear in other places or/and after inlining